app/controllers/api/v0/statistics_controller.rb

```diff
@@ -232,54 +176,18 @@ def plans
     unless Api::V0::StatisticsPolicy.new(@user, :statistics).plans?
       raise Pundit::NotAuthorizedError
     end
-    @org_plans = []
-    @user.org.users.each do |user|
-      user.plans.each do |plan|
-        unless @org_plans.include? plan
-          @org_plans += [plan]
-        end
-      end
-    end
-    @org_plans = restrict_date_range(@org_plans)
-    respond_with @org_plans
+    respond_with @user.org.plans
   end
 
 
   private
 
-  ##
-  # Takes in an array of active_reccords and restricts the range of dates
-  # to those specified in the params
-  #
-  # objects - any active_reccord reccords which have the "created_at" field specified
-  #
-  # Returns Array
-  def restrict_date_range(objects)
-    # set start_date to either passed param, or beginning of time
-    if params[:start_date].blank?
-      start_date = Date.new(0)
-    else
-      start_date = Date.strptime(params[:start_date], "%Y-%m-%d")
-    end
-    # set end_date to either passed param or now
-    if params[:end_date].blank?
-      end_date = Date.today
-    else
-      end_date = Date.strptime(params[:end_date], "%Y-%m-%d")
-    end
-
-    filtered = []
-    objects.each do |obj|
-      # apperantly things can have nil created_at
-      if obj.created_at.blank?
-        if params[:start_date].blank? && params[:end_date].blank?
-          filtered += [obj]
-        end
-      elsif start_date <= obj.created_at.to_date && end_date >= obj.created_at.to_date
-        filtered += [obj]
-      end
-    end
-    filtered
+  # Convert start/end dates in hash to a range of Dates
+  def dates_to_range(hash)
+    today = Date.today
+    start_date = Date.parse(hash.fetch("start_date", today.prev_month.to_date.to_s))
+    end_date = Date.parse(hash.fetch("end_date", today.to_date.to_s)) + 1.day
+    start_date..end_date
```

Comment:
well that's way cleaner than my 30 lines from a couple years ago to restrict the range:)